src/topology/sheaves/stalks.lean

```diff
@@ -109,6 +132,173 @@ lemma stalk_hom_ext (F : X.presheaf C) {x} {Y : C} {f₁ f₂ : F.stalk x ⟶ Y}
   (ih : ∀ (U : opens X) (hxU : x ∈ U), F.germ ⟨x, hxU⟩ ≫ f₁ = F.germ ⟨x, hxU⟩ ≫ f₂) : f₁ = f₂ :=
 colimit.hom_ext $ λ U, by { op_induction U, cases U with U hxU, exact ih U hxU }
 
+/-- If two sections agree on all stalks, they must be equal -/
+lemma section_ext (F : sheaf (Type v) X) (U : opens X) (s t : F.presheaf.obj (op U)) :
+  (∀ x : U, F.presheaf.germ x s = F.presheaf.germ x t) → s = t :=
+begin
+  intro h,
+  -- We use `germ_eq` and the axiom of choice, to pick for every point `x` a neighbourhood
+  -- `V`, such that the restrictions of `s` and `t` to `V` coincide. Since a restriction
+  -- map `V ⟶ U` is *data* (lives in `Type`, not `Prop`), we encode all of this as a dependent
+  -- pair (sigma type)
+  let V : Π (x : U), Σ (V : open_nhds x.1),
+      { iVU : V.1 ⟶ U // F.presheaf.map iVU.op s = F.presheaf.map iVU.op t } := λ x, by {
+    choose V m iVU₀ iVU₁ heq using (F.presheaf.germ_eq x.1 x.2 x.2 s t (h x)),
+    refine ⟨⟨V,m⟩,iVU₀,_⟩,
+    convert heq,
+  },
+  refine F.eq_of_locally_eq' (λ x, (V x).1.1) U (λ x, (V x).2.1) _ s t (λ x, (V x).2.2),
+  -- Here, it remains to show that the choosen neighborhoods really define a cover of `U`
+  intros x hxU,
+  rw [subtype.val_eq_coe, opens.mem_coe, opens.mem_supr],
+  exact ⟨⟨x,hxU⟩,(V ⟨x,hxU⟩).1.2⟩,
+end
+
+@[simp, reassoc] lemma stalk_functor_map_germ {F G : X.presheaf C} (U : opens X) (x : U)
+  (f : F ⟶ G) : germ F x ≫ (stalk_functor C x.1).map f = f.app (op U) ≫ germ G x :=
+colimit.ι_map (whisker_left ((open_nhds.inclusion x.1).op) f) (op ⟨U, x.2⟩)
+
+@[simp] lemma stalk_functor_map_germ_apply (U : opens X) (x : U) {F G : X.presheaf (Type v)}
+  (f : F ⟶ G) (s : F.obj (op U)) :
+  (stalk_functor (Type v) x.1).map f (germ F x s) = germ G x (f.app (op U) s) :=
+congr_fun (stalk_functor_map_germ U x f) s
```

Comment:
Similarly this can also be by `@[elementwise]`.

Reply:
I tried this, unfortunately the generated simp lemmas don't fire where I would like them to. There seems to be some issue with the coercions `coe_fn`, not even `erw` worked. The only way I found was through some pretty annoying unfolding of definitions, before I could rewrite, like this:
```
lemma app_injective_of_stalk_functor_map_injective {F : sheaf (Type v) X} {G : presheaf (Type v) X}
  (f : F.presheaf ⟶ G) (h : ∀ x : X, injective ((stalk_functor (Type v) x).map f)) (U : opens X) :
  injective (f.app (op U)) := λ s t hst,
begin
  apply section_ext,
  intro x,
  apply h x.1,
  have := stalk_functor_map_germ_apply U x f s,
  -- here, I need to explicitly unfold the coercions before I can rewrite...
  dsimp [coe_fn, concrete_category.has_coe_to_fun, forget, concrete_category.forget] at this,
  erw this,
  sorry
end
```

Reply:
Maybe we can try to solve this in a separate PR?

Reply:
Sounds good!